modules/seqexec/model/src/main/scala/seqexec/model/Step.scala

```diff
@@ -80,29 +168,89 @@ object Step {
     }
 
     def isConfiguring: Boolean = s match {
-      case StandardStep(_, _, _, _, _, _, c, _) => c.map(_._2).contains(ActionStatus.Running)
-      case _                                    => false
+      case StandardStep(_, _, _, _, _, _, c, _) =>
+        c.count(_._2 === ActionStatus.Running) > 0
+      case NodAndShuffleStep(_, _, _, _, _, _, c, _) =>
+        c.count(_._2 === ActionStatus.Running) > 0
+      case _ => false
     }
 
-    def isFinished: Boolean = s.status === StepState.Completed || s.status === StepState.Skipped
+    def isFinished: Boolean =
+      s.status === StepState.Completed || s.status === StepState.Skipped
 
     def wasSkipped: Boolean = s.status === StepState.Skipped
 
+    def canRunFrom: Boolean = s.status match {
+      case StepState.Pending | StepState.Failed(_) => true
+      case _                                       => false
+    }
+
+    def canConfigure: Boolean = s.status match {
+      case StepState.Pending | StepState.Paused | StepState.Failed(_) => true
+      case _                                                          => false
+    }
+
   }
 }
 
+@Lenses
 final case class StandardStep(
-  override val id: StepId,
-  override val config: StepConfig,
-  override val status: StepState,
+  override val id:         StepId,
+  override val config:     StepConfig,
+  override val status:     StepState,
   override val breakpoint: Boolean,
-  override val skip: Boolean,
-  override val fileId: Option[dhs.ImageFileId],
-  configStatus: List[(Resource, ActionStatus)],
-  observeStatus: ActionStatus
+  override val skip:       Boolean,
+  override val fileId:     Option[dhs.ImageFileId],
+  configStatus:            List[(Resource, ActionStatus)],
+  observeStatus:           ActionStatus
 ) extends Step
 
 object StandardStep {
-  implicit val equal: Eq[StandardStep] = Eq.by(x => x: Step)
+  implicit val equalStandardStep: Eq[StandardStep] = Eq.by(
+    x =>
+      (x.id,
+       x.config,
+       x.status,
+       x.breakpoint,
+       x.skip,
+       x.fileId,
+       x.configStatus,
+       x.observeStatus)
+  )
 }
+
+@Lenses
+final case class NodAndShuffleStatus(observing: ActionStatus)
+
+object NodAndShuffleStatus {
+
+  implicit val equalNodAndShuffleStatus: Eq[NodAndShuffleStatus] =
+    Eq.by(x => (x.observing))
+}
+
 // Other kinds of Steps to be defined.
+@Lenses
+final case class NodAndShuffleStep(
```

Comment:
Client side representation of a Nod and shuffle step, it needs some more data than present now